ql/src/test/results/clientpositive/llap/alter_partition_coltype.q.out

```diff
@@ -453,6 +453,16 @@ POSTHOOK: Input: default@src
 POSTHOOK: Output: pt@alterdynamic_part_table
 POSTHOOK: Output: pt@alterdynamic_part_table@partcol1=1/partcol2=1
 POSTHOOK: Lineage: alterdynamic_part_table PARTITION(partcol1=1,partcol2=1).intcol EXPRESSION []
+PREHOOK: query: insert into table pt.alterdynamic_part_table partition(partcol1, partcol2) select '2', '2', NULL
+PREHOOK: type: QUERY
+PREHOOK: Input: _dummy_database@_dummy_table
+PREHOOK: Output: pt@alterdynamic_part_table
+POSTHOOK: query: insert into table pt.alterdynamic_part_table partition(partcol1, partcol2) select '2', '2', NULL
+POSTHOOK: type: QUERY
+POSTHOOK: Input: _dummy_database@_dummy_table
+POSTHOOK: Output: pt@alterdynamic_part_table
+POSTHOOK: Output: pt@alterdynamic_part_table@partcol1=2/partcol2=__HIVE_DEFAULT_PARTITION__
```

Comment:
I see that listPartitionNames() is listing the default partition '__HIVE_DEFAULT_PARTITION__'. I think this is not what we expect right? We should ignore this value, right?

Reply:
It's the right behavior, the default partition is ignored by 
https://github.com/apache/hive/blob/71907d883c7317083ec2d2a957a5d03ffe19d725/ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/alter/AlterTableAlterPartitionOperation.java#L100-L101

For this line change, as we insert a null value for partition column `partcol2`, so it will add a new partition `partcol1=2/partcol2=__HIVE_DEFAULT_PARTITION__` to the table `alterdynamic_part_table `.

Reply:
Any ideas? the statement `alter table pt.alterdynamic_part_table partition column (partcol2 int)` changes the string column `partcol2` to int,  and we should ignore all the partition `partcol1=*/partcol2=__HIVE_DEFAULT_PARTITION__`. Does it make sense?

Reply:
Ack